server/src/Service/InteractionService.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace App\Service;
+
+use App\Dao\RevisionDao;
+
+class InteractionService {
+
+	/**
+	 * @var RevisionDao
+	 */
+	private $revisionDao;
+
+	/**
+	 * @param RevisionDao $revisionDao
+	 */
+	public function __construct( RevisionDao $revisionDao ) {
+		$this->revisionDao = $revisionDao;
+	}
+
+	/**
+	 * @param string[] $users
+	 * @param int|null $startDate
+	 * @param int|null $endDate
+	 * @param int $limit
+	 * @param string|null $continue
+	 * @return array
+	 */
+	public function getInteraction(
+		$users, $startDate = null, $endDate = null, $limit = 50, $continue = null
+	) {
+		$this->validateUsers( $users );
+		$this->validateLimit( $limit );
+
+		list( $revisionIds, $continue ) = $this->revisionDao->getUserRevisionsInCommonPages(
+				$users,  $startDate, $endDate, $limit, $continue
+		);
+
+		$revisions = [];
+		if ( $revisionIds ) {
+			$revisions = $this->revisionDao->getRevisionDetails( $revisionIds );
+			$revisions = $this->parseAndMapRevisions( $revisions );
+		}
+
+		return [ $revisions, $continue ];
```

Comment:
Again.. I wonder if it would be better to return a value object than an array.

Reply:
I didn't see any benefits on returning an object over an array in this particular case. What do you suggest?

Reply:
uhh.. I guess there isn't a huge benefit, just looks odd.. but no more odd than all of PHP so... :P So this is fine.